tests/test_formatter.py

```
@@ -2091,3 +2088,789 @@ def test_index_of_first_docstring_match():
     from snakefmt.formatter import index_of_first_docstring
 
     assert index_of_first_docstring('"""docstring"""') == 14
+
+
+class TestFmtOffOn:
+    """Tests for # fmt: off / # fmt: on directives."""
+
+    def test_fmt_off_at_start(self):
+        for code, formatted in (
```

Ruff (0.15.7)

[warning] Loop control variable `formatted` not used within loop body

Rename unused `formatted` to `_formatted`

(B007)